cmd/integration/commands/export_db_data.go

```diff
@@ -0,0 +1,563 @@
+			logger.Error(err.Error())
+			return
+		}
+
+		eventSegments := allBorSnapshots.ViewType(snaptype.BorEvents).VisibleSegments
+		blockReader := freezeblocks.NewBlockReader(allSnapshots, allBorSnapshots)
+		lastFrozenEventId := blockReader.LastFrozenEventId()
+		iterateSnapshots := lastFrozenEventId > 0 && fromNum <= lastFrozenEventId
+		iterateDb := true
+
+		var sb strings.Builder
+		sb.WriteString("source")
+		sb.WriteRune(',')
+		sb.WriteString("id")
+		sb.WriteRune(',')
+		sb.WriteString("time_unix_sec")
+		sb.WriteRune(',')
+		sb.WriteString("date_time_utc")
+		sb.WriteRune(',')
+		sb.WriteString("tx_hash")
+		sb.WriteRune(',')
+		sb.WriteString("log_index")
+		sb.WriteRune('\n')
+		writeEventRow := func(source string, event *heimdall.EventRecordWithTime) {
+			sb.WriteString(source)
+			sb.WriteRune(',')
+			sb.WriteString(strconv.FormatUint(event.ID, 10))
+			sb.WriteRune(',')
+			sb.WriteString(strconv.FormatInt(event.Time.Unix(), 10))
+			sb.WriteRune(',')
+			sb.WriteString(event.Time.UTC().Format(time.RFC3339))
+			sb.WriteRune(',')
+			sb.WriteString(event.TxHash.String())
+			sb.WriteRune(',')
+			sb.WriteString(strconv.FormatUint(event.LogIndex, 10))
+			sb.WriteRune('\n')
+		}
+
+	snapshotLoop:
+		for i := 0; iterateSnapshots && i < len(eventSegments); i++ {
+			seg := eventSegments[i]
+			getter := seg.MakeGetter()
+			for getter.HasNext() {
+				var buf []byte
+				buf, _ = getter.Next(buf)
+
+				eventBytes := rlp.RawValue(libcommon.Copy(buf[length.Hash+length.BlockNum+8:]))
+				var event heimdall.EventRecordWithTime
+				if err := event.UnmarshallBytes(eventBytes); err != nil {
+					logger.Error(err.Error())
+					return
+				}
+
+				if event.ID < fromNum {
+					continue
+				}
+
+				if event.ID >= toNum {
+					iterateDb = false
+					break snapshotLoop
+				}
+
+				writeEventRow("SNAPSHOTS", &event)
+			}
+		}
+
+		c, err := tx.Cursor(kv.BorEvents)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer c.Close()
+		from := make([]byte, 8)
+		binary.BigEndian.PutUint64(from, max(fromNum, lastFrozenEventId+1))
+		var k, v []byte
+		for k, v, err = c.Seek(from); iterateDb && err == nil && k != nil; k, v, err = c.Next() {
+			eventId := binary.BigEndian.Uint64(k)
+			if eventId >= toNum {
+				break
+			}
+
+			var event heimdall.EventRecordWithTime
+			if err := event.UnmarshallBytes(v); err != nil {
+				logger.Error(err.Error())
+				return
+			}
+
+			writeEventRow("DB", &event)
+		}
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		if err := os.WriteFile(outputCsvFile, []byte(sb.String()), 0600); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+	},
+}
+
+var cmdExportHeimdallEventsPerBlock = &cobra.Command{
+	Use:   "export_heimdall_events_per_block",
+	Short: "",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := debug.SetupCobra(cmd, "integration")
+		db, err := openDB(dbCfg(kv.ChainDB, datadirCli+"/chaindata"), true, logger)
```

Comment:
Maybe we should use `path.Join` instead of `datadirCli+"/chaindata"`

Reply:
@shohamc1 changed to path.Join